core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/handler/BaseAsyncClientHandler.java

```diff
@@ -120,22 +171,20 @@ protected BaseAsyncClientHandler(SdkClientConfiguration clientConfiguration,
                 finalizeSdkHttpRequestContext.requestBody().isPresent()) {
                 marshalled = marshalled.toBuilder()
                                        .contentStreamProvider(
-                                           finalizeSdkHttpRequestContext.requestBody().get().contentStreamProvider())
+                                           finalizeSdkHttpRequestContext.requestBody()
+                                                                        // To appease static-code analysis
+                                                                        .orElse(null)
+                                                                        .contentStreamProvider())
```

Comment:
This seems hacky and easy to get an exception

Reply:
Yes I was trying to appease the static analysis gods and made it worse. I'm going to revert.